ssl/ssl_lib.c

```diff
@@ -6133,41 +6136,40 @@ static int ct_extract_ocsp_response_scts(SSL_CONNECTION *s)
 {
 # ifndef OPENSSL_NO_OCSP
     int scts_extracted = 0;
-    const unsigned char *p;
     OCSP_BASICRESP *br = NULL;
     OCSP_RESPONSE *rsp = NULL;
     STACK_OF(SCT) *scts = NULL;
-    int i;
+    int i, j;
 
-    if (s->ext.ocsp.resp == NULL || s->ext.ocsp.resp_len == 0)
+    if (s->ext.ocsp.resp_ex == NULL)
         goto err;
 
-    p = s->ext.ocsp.resp;
-    rsp = d2i_OCSP_RESPONSE(NULL, &p, (int)s->ext.ocsp.resp_len);
-    if (rsp == NULL)
-        goto err;
+    for (j = 0; j < sk_OCSP_RESPONSE_num(s->ext.ocsp.resp_ex); j++) {
+        rsp = sk_OCSP_RESPONSE_value(s->ext.ocsp.resp_ex, j);
+        if (rsp == NULL)
+            goto err;
 
-    br = OCSP_response_get1_basic(rsp);
-    if (br == NULL)
-        goto err;
+        br = OCSP_response_get1_basic(rsp);
+        if (br == NULL)
+            goto err;
 
-    for (i = 0; i < OCSP_resp_count(br); ++i) {
-        OCSP_SINGLERESP *single = OCSP_resp_get0(br, i);
+        for (i = 0; i < OCSP_resp_count(br); ++i) {
+            OCSP_SINGLERESP *single = OCSP_resp_get0(br, i);
 
-        if (single == NULL)
-            continue;
+            if (single == NULL)
+                continue;
 
-        scts =
-            OCSP_SINGLERESP_get1_ext_d2i(single, NID_ct_cert_scts, NULL, NULL);
-        scts_extracted =
-            ct_move_scts(&s->scts, scts, SCT_SOURCE_OCSP_STAPLED_RESPONSE);
-        if (scts_extracted < 0)
-            goto err;
+            scts = OCSP_SINGLERESP_get1_ext_d2i(single,
+                                                NID_ct_cert_scts, NULL, NULL);
+            scts_extracted = ct_move_scts(&s->scts, scts,
+                                          SCT_SOURCE_OCSP_STAPLED_RESPONSE);
+            if (scts_extracted < 0)
+                goto err;
+        }
```

Comment:
Missing  call to ` SCT_LIST_free(scts);    OCSP_BASICRESP_free(br);` inside the loop.
May be `scts_extracted` should sum up, so it can return the count. 
Take care to `scts_extracted` negative value on error.

Reply:
I moved the calls of 
```
SCT_LIST_free(scts);
OCSP_BASICRESP_free(br);
```
into the loop. The memory is freed before the next loop.